plugin/src/main/java/com/denizenscript/denizen/events/block/BlockExplodesScriptEvent.java

```diff
@@ -35,12 +35,45 @@ public class BlockExplodesScriptEvent extends BukkitScriptEvent implements Liste
     //
     // @Determine
     // ListTag(LocationTag) to set a new lists of blocks that are to be affected by the explosion.
-    // ElementTag(Decimal) to change the strength of the explosion.
+    // "STRENGTH:<ElementTag(Decimal)>" to change the strength of the explosion.
     //
     // -->
 
     public BlockExplodesScriptEvent() {
         registerCouldMatcher("<block> explodes");
+        this.<BlockExplodesScriptEvent, ObjectTag>registerOptionalDetermination(null, ObjectTag.class, (evt, context, value) -> {
+            if (value.toString().contains(",") || value.toString().startsWith("li@")) { // Raw ListTag check due to there not being a prefix for block strength previously
+                evt.event.blockList().clear();
+                boolean valid = false;
+                for (String newBlock : ListTag.valueOf(value.toString(), context)) {

```

Comment:
Should use `value.asType(ListTag.class)`, and then can `#filter(LocationTag.class)` on the list instead of the manual error checking.